### PR TITLE
Add bigint to the primitive type

### LINKED ISSUE #86: BigInt Support
## Is your feature request related to a real problem or use-case?
Adding BigInt to the Primitive type as [its slated to become one](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt).

## Describe a solution including usage in code example
```ts
type Primitive = number | boolean | string | symbol | bigint
```

## Who does this impact? Who is this for?
Users using the Stage 3 Proposal of Bigint and [Typescript >=3.2](https://devblogs.microsoft.com/typescript/announcing-typescript-3-2/)

## Describe alternatives you've considered (optional)
As I'm not sure how good support for the (JS, not the TS kind of 'overload') typeof operator of eventually employed polyfills is, and how older TS implementations behave with this unknown type, we might want to hold back on that for now, or guard this implementation.

## Additional context (optional)
[Partially Linked in the text before]
* [Microsofts devblog to TS 3.2 which added BigInt support](https://devblogs.microsoft.com/typescript/announcing-typescript-3-2/)
* [MDN entry for BigInt](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt)
* MDN entry for typeof](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/typeof)

### PR DESCRIPTION
**Pull request by 9SMTM6**
## Description
Added bigint to the `Primitive` type.

I've tested an import of the `Primitive` type with bigint into an project using TS 3.1.6 with default settings, and it worked - of course it didnt recognize bigint, but no syntax error on usage on other variables and it didnt break support for the other types. Thus I'm not creating another version of this package for this. But bigint support was only added in TS 3.2, which is below the required v3.1 you ask for for v3.X of this package, so make some more tests (i.e. with different tsconfig settings) at your own discretion.

Also I didnt even increase the minor version as I couldnt find any documentation on whether I should.

Note: I had to change the syntax of the node scripts slightly to make them work on my PC. The changes are in a single commit you can revert if thats an issue for you.

Also the type tests dont really seem to do much at all... In one of my experiments I have tried [`tsd`](https://www.npmjs.com/package/tsd) and it worked so far, if that doesnt make too many problems with integration in the pipeline, you might want to check it out.

## Related issues:
- Resolved #86

## Checklist

* [x] I have read [CONTRIBUTING.md](../CONTRIBUTING.md)
* [x] I have linked all related issues above
* [x] I have rebased my branch

### CODE REVIEW

File: src/mapped-types.ts [hunk 1]
```diff
@@ -5,14 +5,14 @@
 
 /**
  * Primitive
- * @desc Type representing primitive types in TypeScript: `number | boolean | string | symbol`
+ * @desc Type representing primitive types in TypeScript: `number | bigint | boolean | string | symbol`
  * @example
  *   type Various = number | boolean | string | symbol | object;
```
Comment (piotrwitek) [inline src/mapped-types.ts]: Could you please update example as well @9SMTM6 ?

Comment (9SMTM6) [reply src/mapped-types.ts]: If you wish so. My reason for not changing it:

The example is still correct as it is currently, you just exclude a type that was never there. Maybe it causes an syntax error, I'm not that familiar with your `Exclude` type, but I wouldn't think so.

And if it works like that - as I'd strongly suspect - the way it is it demonstrates that you don't in fact need to be careful with excluding primitives from any type (that the type from which to exclude doesn't have to be either a complete superset of primitive, but can have any subset, including empty, of primitive).

In fact, I'd maybe even suggest stripping some more primitive types from the example.

Comment (piotrwitek) [reply src/mapped-types.ts]: Ok, I like your updated suggestion, go for it! Thanks

**Comment by piotrwitek** [comment]
Great job, thanks for contribution @9SMTM6 👍

**Pull request merged**